platform/mellanox/mlnx-fw-upgrade.j2

```diff
@@ -339,7 +427,13 @@ function UpgradeFW() {
     if [[ "${_FW_CURRENT}" == "${_FW_AVAILABLE}" ]]; then
         ExitSuccess "firmware is up to date"
     else
+        if [[ "${DRY_RUN}" == "${YES_PARAM}" ]]; then
```

Comment:
Why do we need this? Just for user convenience?

Reply:
Got it, used it in bfnet.sh. I think "upgrade_check" or similar name would be better than dry_run

Reply:
I don't agree with this. `--dry-run` is widely used by Linux commands for purposes like this.